lib/internal/fs/streams.js

```diff
@@ -286,10 +348,11 @@ function WriteStream(path, options) {
 
   // Path will be ignored when fd is specified, so it can be falsy
   this.path = toPathIfFileURL(path);
-  this.fd = options.fd === undefined ? null : options.fd;
   this.flags = options.flags === undefined ? 'w' : options.flags;
   this.mode = options.mode === undefined ? 0o666 : options.mode;
 
+  importFd(this, options);
```

Comment:
```suggestion
  ReflectApply(FIleHandleCompatibleStream, this, [options]);
```

Reply:
I don't understand what would this accomplish, you would still see `ReadStream`/`WriteStream` as a class name in the debugger?

Reply:
Yes you would.
It’s an unusual thing to have a function mutating its parameters (correct me if I’m wrong, but I don’t think it happens anywhere else in the code base), however having a parent class is quite common. I think it’s more readable with my suggestion.

Reply:
If I understand correctly (`options.fd` being one of several types), there are plenty, look for
`/ERR_INVALID_ARG_TYPE.*\['.*,/`

Reply:
Hum, I don't get what you mean, my objection is not about `options.fd` being of several types. Let me try to rephrase: my issue with the PR is that `importFd` mutates its parameters: it defines an `fd` property on its first parameter `stream`, and sometimes other properties as well. That's not ideal, because for someone in the future reading the code, it would probably not be obvious to them that those properties are defined at all. I don't think there is any use of a function that mutates its parameters in the code base, and I'm pretty sure we would prefer to avoid it if we can.

Reply:
Most functions that accept a stream as a first argument do change its state. They are in fact member-like functions of both classes. `close` which is just above `importFd` modifies its state too.

Reply:
OK, my objection remains (I still think a parent class would be preferable), but I'm not blocking this PR from landing. I'll open a follow up PR to propose my changes once this one has landed.